File: frontend/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  category: string;
  description: string;
  price: number;
  imageUrl?: string;
  sortOrder: number;
  createdAt: string;
}

export type CreateProductDto = Omit<Product, 'id' | 'sortOrder' | 'createdAt'>;
export type UpdateProductDto = Omit<Product, 'id' | 'sortOrder' | 'createdAt'>;

export interface ReorderItem {
  id: string;
  sortOrder: number;
}

File: backend/Models/Product.cs
namespace ProductCatalog.Api.Models;

public class Product
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public required string Name { get; set; }
    public required string Category { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string? ImageUrl { get; set; }
    public int SortOrder { get; set; }
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
}

public class CreateProductDto
{
    public required string Name { get; set; }
    public required string Category { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string? ImageUrl { get; set; }
}

public class UpdateProductDto
{
    public required string Name { get; set; }
    public required string Category { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string? ImageUrl { get; set; }
}

public class ReorderItem
{
    public Guid Id { get; set; }
    public int SortOrder { get; set; }
}
